js/views/userPageVw.js

```diff
@@ -255,40 +256,55 @@ module.exports = Backbone.View.extend({
-            }
+        if (self.isRemoved()) return;
+
+        if (response.profile){
+          $('.js-loadingModal').addClass('hide');
+          if (self.options.ownPage === true){
+            model.set('headerURL', self.options.userModel.get('serverUrl') + "get_image?hash=" + model.get('profile').header_hash);
+            model.set('avatarURL', self.options.userModel.get('serverUrl') + "get_image?hash=" + model.get('profile').avatar_hash);
           }else{
-            //model was returned as a blank object
-            $('.js-loadingModal').addClass('hide');
-            showErrorModal(window.polyglot.t('errorMessages.getError'), window.polyglot.t('errorMessages.userError') + "<br/><br/>" + self.pageID);
+            model.set('headerURL', self.options.userModel.get('serverUrl') + "get_image?hash=" + model.get('profile').header_hash + "&guid=" + self.pageID);
+            model.set('avatarURL', self.options.userModel.get('serverUrl') + "get_image?hash=" + model.get('profile').avatar_hash + "&guid=" + self.pageID);
           }
+        }else{
+          //model was returned as a blank object
+          $('.js-loadingModal').addClass('hide');
+          messageModal.show(window.polyglot.t('errorMessages.storeUnavailable'), window.polyglot.t('errorMessages.userError') + "<br/><br/>" + self.pageID);
+          self.bindModalCloseHandler();
+        }
 
-          // Cache user avatar in localStorage
-          var profile = model.toJSON().profile;
-          window.localStorage.setItem("avatar_" + self.pageID, profile.avatar_hash);
+        // Cache user avatar in localStorage
+        var profile = model.toJSON().profile;
+        window.localStorage.setItem("avatar_" + self.pageID, profile.avatar_hash);
 
-          self.model.set({user: self.options.userModel.toJSON(), page: model.toJSON()});
-          self.model.set({ownPage: self.options.ownPage});
-          self.render();
-        }
+        self.model.set({user: self.options.userModel.toJSON(), page: model.toJSON()});
+        self.model.set({ownPage: self.options.ownPage});
+        self.render();
       },
       error: function(model, response){
+        if (self.isRemoved()) return;
+
         $('.js-loadingModal').addClass('hide');
-        showErrorModal(window.polyglot.t('errorMessages.getError'), window.polyglot.t('errorMessages.userError') + "<br/><br/>" + self.pageID);
+        messageModal.show(window.polyglot.t('errorMessages.storeUnavailable'), window.polyglot.t('errorMessages.userError') + "<br/><br/>" + self.pageID);
+        self.bindModalCloseHandler();
         self.model.set({user: self.options.userModel.toJSON(), page: {profile: ""}});
         self.render();
       }
     });
   },
 
+  bindModalCloseHandler: function(e) {
+    if (this.modalCloseHandler) return;
+
+    this.modalCloseHandler = function() {
+      window.history.back();
```

Comment:
This doesn't account for the situation where if you start the app with the saved last state pointing at a user profile that is now offline, a blank user view is still shown.

Can we check to the length of history here, and if there's nothing to go back to load the home view instead?